src/core/control/PdfCache.cpp

```diff
@@ -35,27 +38,31 @@ class PdfCacheEntry {
     cairo_surface_t* rendered;
 };
 
-PdfCache::PdfCache(size_t size) { this->size = size; }
+PdfCache::PdfCache(const XojPdfDocument& doc, size_t size): pdfDocument(doc), size(size) {}
 
 PdfCache::~PdfCache() {
     clearCache();
     this->size = 0;
 }
 
-void PdfCache::setZoom(double zoom) { this->zoom = zoom; }
-
 void PdfCache::setRefreshThreshold(double threshold) { this->zoomRefreshThreshold = threshold; }
 
-void PdfCache::setAnyZoomChangeCausesRecache(bool b) { this->zoomClearsCache = b; }
+void PdfCache::setMaxSize(size_t newSize) {
+    this->size = newSize;
+    while (this->data.size() > this->size) {
+        delete this->data.back();
+        this->data.pop_back();
+    }
+}
 
 void PdfCache::clearCache() {
     for (PdfCacheEntry* e: this->data) { delete e; }
     this->data.clear();
 }
 
-auto PdfCache::lookup(const XojPdfPageSPtr& popplerPage) -> PdfCacheEntry* {
+auto PdfCache::lookup(size_t pdfPageNo) const -> PdfCacheEntry* {
     for (PdfCacheEntry* e: this->data) {
```

Comment:
For a follow up PR: This looks rather inefficient given that this is the lookup method on a _cache_, should probably change `data` to a map of some sort.

Reply:
Well, it actually turns out the cache size is (always) 10. The setting parameter we fetch by calling `control->getSettings()->getPdfPageCacheSize()` has only a default value, and no way of changing it (besides maybe editing the config file by hand).

With those kind of sizes, a map would probably actually be slower